sansio_lsp_client/structs.py

```diff
@@ -409,21 +413,26 @@ class ProgressValue(BaseModel):
 class WorkDoneProgressValue(ProgressValue):
     pass
 
+class MWorkDoneProgressKind(enum.Enum):
+    BEGIN = 'begin'
+    REPORT = 'report'
+    END = 'end'
+
 class WorkDoneProgressBeginValue(WorkDoneProgressValue):
-    kind: str = 'begin'
+    kind: MWorkDoneProgressKind # .BEGIN
```

Comment:
I think it's possible to use `Literal` with enums.

Reply:
> [New in version 3.8.](https://docs.python.org/3/library/typing.html#typing.Literal)

Reply:
`from typing_extensions import Literal`

Reply:
Good to know :)